batch.go

```diff
@@ -991,3 +1004,121 @@ func (batch *Batch) Equal(other Batcher) bool {
 	}
 	return len(batch.Entries) == equalEntries && equalEntries != 0
 }
+
+// WithOffset sets the Offset information onto a Batch so that during Create a "balanced offset"
+// record is appended at the end.
+//
+// Balanced offset records sum a File's Debits or Credits to zero in a single transaction.
+// Offset files are used to offset transactions from a single account inside of the ODFI.
+//
+// If the Batch contains Debits the offset record will be a Credit and if the batch contains Credits
+// the offset record will be a debit.
+func (b *Batch) WithOffset(off *Offset) {
+	b.offset = off
+}
+
+func (b *Batch) upsertOffset() error {
+	if b == nil || b.offset == nil {
+		return nil
+	}
+
+	// remove any Offset records already on the batch
+	addOffset := true // append to EntryAddendaCount
+	for i := 0; i < len(b.Entries); i++ {
+		// TODO(adam): Should we remove this based on checking the last element is
```

Comment:
Can the TODO be moved to a GIthub issue as a question or enhancement

Reply:
Done - https://github.com/moov-io/ach/issues/540